File: SNS24.API/Models/Doctor/Doctor.cs
using SNS24.WebApi.Models.Common;

namespace SNS24.WebApi.Models
{
    public class Doctor : ApplicationUser
    {
        public string LicenseNumber { get; set; }
        public string Specialty { get; set; }
        public ICollection<Institution> Institutions { get; set; } = new HashSet<Institution>();
        public ICollection<MedicalLeave> MedicalLeavesIssued { get; set; } = new HashSet<MedicalLeave>();

        public Doctor()
        {
            Role = Enums.Role.Doctor;
        }
    }
}

File: webapp/src/app/shared/interfaces/doctor/doctor.ts
import { ApplicationUser } from '../user/applicationUser';
import { Institution } from '../institution';
import { MedicalLeave } from '../medical-appointment/medical-leave';

export interface Doctor extends ApplicationUser {
  licenseNumber?: string;
  specialty?: string;
  isPublicSector?: boolean;
  institutions?: Institution[] | null;
  medicalLeavesIssued?: MedicalLeave[];
}
